src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/AsyncLazy`1.cs

```diff
@@ -350,12 +338,12 @@ public Task<T> GetValueAsync(CancellationToken cancellationToken)
     private AsynchronousComputationToStart RegisterAsynchronousComputation_NoLock()
     {
         Contract.ThrowIfTrue(_computationActive);
-        Contract.ThrowIfNull(_asynchronousComputeFunction);
+        Contract.ThrowIfFalse(HasAsynchronousComputeFunction);
 
         _asynchronousComputationCancellationSource = new CancellationTokenSource();
         _computationActive = true;
 
-        return new AsynchronousComputationToStart(_asynchronousComputeFunction, _asynchronousComputationCancellationSource);
+        return new AsynchronousComputationToStart(InvokeComputeFunctionAsync, _asynchronousComputationCancellationSource);
```

Comment:
⚠️ This is now going to allocate a delegate at this location

Reply:
Hmm, not immediately obvious to me how to work around this. Any suggestions?